db/oracle_schema.rb

```diff
@@ -639,13 +639,13 @@
     t.datetime "paid_at",                          precision: 6
     t.datetime "created_at",                       precision: 6
     t.datetime "updated_at",                       precision: 6
-    t.datetime "due_on",                           precision: 6
+    t.date     "due_on"
     t.string   "pdf_file_name"
     t.string   "pdf_content_type"
     t.integer  "pdf_file_size",                    precision: 38
     t.datetime "pdf_updated_at",                   precision: 6
-    t.datetime "period",                           precision: 6
-    t.datetime "issued_on",                        precision: 6
+    t.date     "period"
```

Comment:
Those are really stored as DB `DATE`  in MySQL, Oracle and Postgres ....
Oracle adapter was just buggy, as it could not differentiate `DATE` and `TIMESTAMPS` before 1.7

No need to convert them for oracle as they are already in the right type`DATE`.